spec/fortify_spec.rb

```diff
@@ -68,74 +68,101 @@
-        project = Project.first
-        expect(Project.can?(:read, :created_at)).to eq false
-        expect(Project.can?(:read, :text)).to eq true
-        expect(project.can?(:read, :created_at)).to eq false
-        expect(project.can?(:read, :text)).to eq true
+      context '#permitted_attributes_for_read' do
+        it 'cannot read unreadable attributes' do
+          project = Project.first
+          expect(Project.can?(:read, :created_at)).to eq false
+          expect(Project.can?(:read, :text)).to eq true
+          expect(project.can?(:read, :created_at)).to eq false
+          expect(project.can?(:read, :text)).to eq true
+        end
       end
-    end
 
-    context 'applying validation' do
-      it 'allows permitted actions' do
-        project = Project.first
+      context 'applying validation' do
+        it 'allows permitted actions' do
+          project = Project.first
 
-        expect(project.can?(:update, :name)).to eq true
-        project.name = 'ver2'
-        expect(project.valid?).to eq true
-      end
+          expect(project.can?(:update, :name)).to eq true
+          project.name = 'ver2'
+          expect(project.valid?).to eq true
+        end
 
-      it 'does not allow unpermitted updates' do
-        project = Project.first
+        it 'does not allow unpermitted updates' do
+          project = Project.first
 
-        expect(project.can?(:update, :created_at)).to eq false
-        project.created_at = DateTime.now
-        expect(project.valid?).to eq false
-        expect(project.errors.first[0]).to eq :created_at
-        expect(project.errors.first[1]).to eq "You are not authorizated to perform this action"
-      end
+          expect(project.can?(:update, :created_at)).to eq false
+          project.created_at = DateTime.now
+          expect(project.valid?).to eq false
+          expect(project.errors.first[0]).to eq :created_at
+          expect(project.errors.first[1]).to eq "You are not authorizated to perform this action"
+        end
+
+        it 'does not allow updates for unauthorized resources' do
+          expect(default_user.can?(:update)).to be true
+          expect(other_user.can?(:update)).not_to be true
+        end
+
+        it 'does not allow unpermitted destroys' do
+          project = Project.first
+          expect(project.destroy).to eq false
+        end
 
-      it 'does not allow updates for unauthorized resources' do
-        expect(default_user.can?(:update)).to be true
-        expect(other_user.can?(:update)).not_to be true
+        context 'when the user can perform destroy' do
+          let(:admin_user) { User.find_by(name: 'admin-user') }
+
+          it 'allows destroying' do
+            Fortify.user = admin_user
+
+            project = Project.first
+            expect(project.can?(:destroy)).to eq true
+          end
+        end
       end
+    end
 
-      it 'does not allow unpermitted destroys' do
+    context "when user is not set" do
```

Comment:
We should have a test case here to make sure `user.can?(:update)` also raises the error

Reply:
@brianisme do you think that's a better place to raise the error? in the `can?` method?

Reply:
We can do this:
```
def can?(action, field=nil)
  raise InvalidUserError.new("Fortify user not set") if !Fortify.user

  policy.can?(action, field)
end
```

Reply:
We can also raise it `policy.can?`.

Reply:
`policy.can?` feels like a better idea. So that the logic is confined inside of policy base instead of ActiveRecord base.

Reply:
@brianisme good point. thanks!